packages/rum-core/src/browser/performanceCollection.ts

```diff
@@ -268,11 +268,11 @@ function computeRelativePerformanceTiming() {
   const result: Partial<IndexedPerformanceTiming> = {}
   const timing = performance.timing as IndexedPerformanceTiming
   for (const key in timing) {
-    if (isNumber(timing[key])) {
-      result[key] = timing[key] === 0 ? 0 : getRelativeTime(timing[key] as number)
+    if (isNumber<TimeStamp>(timing[key])) {
+      result[key] = timing[key] === 0 ? 0 : getRelativeTime(timing[key])
     }
   }
-  return result as PerformanceTiming
+  return result as Omit<RumPerformanceResourceTiming, 'entryType'> & Omit<RumPerformanceNavigationTiming, 'entryType'>
```

Comment:
I don't understand this cast, as `RumPerformanceResourceTiming` contains fields that are not in result
```
initiatorType: string
name: string
```

I think we want to return a type like
```
type TypedPerformanceTiming = {
  [key in keyof PerformanceTiming]: TimeStamp
}
```

Reply:
We are using performance.timing to manually generate:
- a resource entry, for the request timings of the initial doc
- a navigation entry, if it is not supported through the observer
But you are right, we can probably do something better here, I'll look into it